test/buy-tshirt.spec.ts

```diff
@@ -1,38 +1,55 @@
+import { MenuContentPage, AddressStepPage, BankPaymentPage, ProductAddedPage,
+  ProductListPage, SummaryContentPage, SignInContent, OrderSummaryPage,
+  ShippingStepPage, PaymentStepPage } from '../src/page';
 
 describe('Buy a t-shirt', () => {
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const addressContentPage: AddressStepPage = new AddressStepPage();
+  const bankContentPage: BankPaymentPage = new BankPaymentPage();
+  const productAddedPage: ProductAddedPage = new ProductAddedPage();
+  const productListPage: ProductListPage = new ProductListPage();
+  const summaryContentPage: SummaryContentPage = new SummaryContentPage();
+  const singInContentPage: SignInContent = new SignInContent();
+  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
 
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
     await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    // product-added-modal.page
+    await productAddedPage.addProduct();
     await(browser.sleep(3000));
-    await $('[style*="display: block;"] .button-container > a').click();
+    // product-list.page
+    await productListPage.checkProductList();
     await(browser.sleep(3000));
-    await $('.cart_navigation span').click();
+    // summary-step.page
+    await summaryContentPage.goToSummary();
     await(browser.sleep(3000));
-
-    await $('#email').sendKeys('aperdomobo@gmail.com');
-    await $('#passwd').sendKeys('WorkshopProtractor');
-    await $('#SubmitLogin > span').click();
+    // sign-in-step.page
+    await singInContentPage.goTosignIn();
     await(browser.sleep(3000));
-
-    await $('#center_column > form > p > button > span').click();
+    // address-step.page
+    await addressContentPage.confirmAddress();
     await(browser.sleep(3000));
-
-    await $('#cgv').click();
+    // order-summary.page
+    await orderSummaryPage.checkOrderSummary();
     await(browser.sleep(3000));
-
-    await $('#form > p > button > span').click();
+    // shipping-step.page
+    await shippingStepPage.goToShippingStep();
     await(browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+    // payment-step.page
+    await paymentStepPage.goToPaymentStep();
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+    // bank-payment.page
+    await bankContentPage.checkBankPayment();
     await(browser.sleep(3000));
 
     await expect($('#center_column > div > p > strong').getText())
```

Comment:
Here you should implement the orderSummary step. That page object should return the title, and here in the test class you should make the verification with the expected title. 

The reason for this is that the idea of the POM is to model the business behavior of an app component, however, the expectations or validations should remain in the test classes. That being said, the right pattern here should be to have a method to retrieve the orderConfirmation from the summary and compare that confirmation with the expected message on the test class.

Reply:
Got it!!